### PR TITLE
[Merged by Bors] - vm: instrument with metrics to track health and perf

### LINKED ISSUE #3270: vm: observability
TBD

### PR DESCRIPTION
**Pull request by dshulyak**
closes: https://github.com/spacemeshos/go-spacemesh/issues/3270

### CODE REVIEW

File: genvm/metrics.go [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+		"block_duration_persist",
+		namespace,
+		"Duration in ns to write updates on disk.",
+		[]string{},
+		prometheus.ExponentialBuckets(10_000_000, 2, 10),
+	).WithLabelValues()
+)
+
+var (
+	stagingCacheSize = metrics.NewGauge(
```
Comment (countvonzero) [inline genvm/metrics.go]: unused?

**Review by countvonzero — APPROVED** [review]

**Comment by dshulyak** [comment]
bors merge

File: genvm/metrics.go (hunk 1 above)
Comment (dshulyak) [reply genvm/metrics.go]: thank, it was basically the same as one below so i removed it

**Comment by dshulyak** [comment]
bors merge

**Comment by dshulyak** [comment]
bors merge

**Comment by dshulyak** [comment]
bors merge

**Pull request closed without merging**